src/main.cpp

```diff
@@ -35,8 +36,22 @@ void publishMessage() {
     if(DEBUG_MEM){
         DEBUG_SERIAL_LN("\nFREE RAM: " + String(System.freeMemory()) + "B / 128000B");
     }
+}
 
+// allows us to define behavior depending on publish status
+void publishCallback(String payload, DataQueue::PublishStatus status) {
```

Comment:
I think we should still publish the Message in the maxFrequency case, can you add that? You could probably also combine this and the publishMessage() function, but that's more a matter of personal preference.

Reply:
Hey yeah I should have said that the implementation of publishCallBack wasn't final.  I was actually hoping to discuss how each case should be handled

Reply:
I'm wondering if it would be worth implementing something in DataQueue to handle the BufferOverflowing case, or if we should just not publish and reset the JsonWriter buffer